extra runner config options

jonnymccullagh opened the pull request:
In our environment we use some gitlab-runner config parameters that are not yet in this ansible role. Example:

```
log_format = "json"
[runners.docker]
disable_cache = false
dns_search = ["us-east-2.somedomain.tld"]
oom_kill_disable = true
```

This PR aims to add those options.

Arffeh commented:
This should probably also have a default set, even if blank. 
The lack of a default in this MR (nor documentation) breaks the "Add log_format to config" step.


```
fatal: [gitlab-runner-1]: FAILED! => {"msg": "The conditional check 'gitlab_runner_log_format | length > 0' failed. The error was: error while evaluating conditional (gitlab_runner_log_format | length > 0): 'gitlab_runner_log_format' is undefined
The error appears to be in '/tmp/awx_47_awjtmtx_/requirements_roles/riemers.gitlab-runner/tasks/global-setup.yml': line 42, column 3, but may
be elsewhere in the file depending on the exact syntax problem.
The offending line appears to be:
- name: Add log_format to config
  ^ here\n"}
```

fourstepper commented:
I am not sure that this feature works as expected - the result for us is this:

`ERROR: Job failed (system failure): prepare environment: Error response from daemon: OCI runtime create failed: sysctl "log_format" is not in a separate kernel namespace: unknown (exec.go:57:0s). Check https://docs.gitlab.com/runner/shells/index.html#shell-profile-loading for more information`

The full config:
```
[[runners]]
  name = "runner.example.com"
  limit = 0
  output_limit = 4096
  url = "https://gitlab.example.com"
  environment = []
  token = "token"
  executor = "docker"
  [runners.custom_build_dir]
  [runners.cache]
    [runners.cache.azure]
  [runners.docker]
    "tls_verify" = false
    image = "debian"
    privileged = true
    disable_entrypoint_overwrite = false
    oom_kill_disable = false
    disable_cache = false
    volumes = ["/var/run/docker.sock:/var/run/docker.sock", "/cache", "/builds:/builds:rw"]
    shm_size = 0
    [runners.docker.sysctls]
      "net.ipv4.ip_forward" = "1"
log_format = "runner"
```

jonnymccullagh commented:
@fourstepper The placement of the log_format line is due to the regexp which tries to place it after 'listen_address'. Since you don't have a 'listen_address' line it is placed it at the EOF. 

@Arffeh yes I should have added a default of 'runner' 

I hope I have addressed these issues with this PR:
https://github.com/riemers/ansible-gitlab-runner/pull/169

jonnymccullagh commented:
@fourstepper @Arffeh Reimer has merged so please test and let me know if anything further needs done
